input/fsh/models.fsh

```diff
@@ -54,13 +54,16 @@ criteria are defined by FHIRPath expressions.
 """
 * resourceVersion from http://hl7.org/fhir/ValueSet/FHIR-version
 * constant 0..* BackboneElement "Constant that can be used in FHIRPath expressions" """
-  A constant is a string that is injected into a FHIRPath expression through the use of a FHIRPath 
+  A constant is a value that is injected into a FHIRPath expression through the use of a FHIRPath
   external constant with the same name.
 """
   * name 1..1 string "Name of constant (referred to in FHIRPath as %[name])"
   * name obeys sql-name
-  * value 1..1 string "Value of constant" """
-    The string that will be substituted in place of the constant reference
+  * value[x] 1..1 string or integer or code "Value of constant" """
```

Comment:
This looks good structurally. But do we need a choice type? They can be difficult to use when implementation. Could the FHIRPath just return the string or code or int?

Reply:
Ah, never-mind I see this is not an expression.

Reply:
I think Ryan is saying that the content is not FHIRPath, just a simple primitive. And if it is, it should include boolean

Reply:
The main thing I like about the choice type is it makes the user more explicit, and leaves us an avenue to support other types of constants in the future. (Specifically, I could see a very strong case to include coding constants, or maybe even quantity constants.)

Reply:
Maybe add all the primitive types?

Reply:
Good call on the Boolean, Grahame. I'll add that quickly.

Reply:
I went ahead and updated the PR to include all primitive types. (It seems unlikely that that many of these will be needed, but any subset here seems arbitrary to me.)